contrib/opencensus-ext-azure/opencensus/ext/azure/extension/azure_functions.py

```diff
@@ -0,0 +1,154 @@
+INSTRUMENTATION_KEY_PREFIX = 'InstrumentationKey='
+
+
+class OpenCensusExtension(AppExtensionBase):
+    """Extension for Azure Functions integration to export traces into Azure
+    Monitor. Ensure the following requirements are met:
+    1. Azure Functions version is greater or equal to v3.0.15584
+    2. App setting PYTHON_ENABLE_WORKER_EXTENSION is set to 1
+    3. OpenCensusExtension.configure() method is called
+    """
+
+    @classmethod
+    def init(cls):
+        cls._tracers: Dict[str, Tracer] = {}
+        cls._exporter: Optional[AzureExporter] = None
+        cls._has_configure_called: bool = False
+        cls._trace_integrations: List[str] = []
+        cls._default_instrumentation_key: Optional[str] = os.environ.get(
+            DEFAULT_INSTRUMENTATION_KEY_SETTING
+        )
+
+    @classmethod
+    def configure(cls,
+                  libraries: List[str],
+                  connection_string: Optional[str] = None,
+                  *args,
+                  **kwargs):
+        """Configure libraries for integrating into OpenCensus extension.
+        Initialize an Azure Exporter that will write traces to AppInsights.
+
+        :type libraries: List[str]
+        :param libraries: the libraries opencensus-ext-* that need to be
+            integrated into OpenCensus tracer. (e.g. ['requests'])
+        :type connection_string: Optional[str]
+        :param connection_string: the connection string for azure exporter
+            to write into. If this is set to None, the extension will write to
+            the AppInsights resource defined in APPINSIGHTS_INSTRUMENTATIONKEY
+        """
+        cls._has_configure_called = True
+
+        cls._trace_integrations = config_integration.trace_integrations(
+            libraries
+        )
+
+        instrumentation_key = cls._parse_ikey_from_cstring(connection_string)
+        if not instrumentation_key and not cls._default_key:
+            raise FunctionExtensionException(
+                'Please ensure either connection_string is passed into '
+                'OpenCensusExtension.configure() method, or the app setting '
+                'APPINSIGHTS_INSTRUMENTATIONKEY is provided.'
+            )
+
+        cls._exporter = AzureExporter(
+            instrumentation_key=instrumentation_key or cls._default_key
+        )
+
+    @classmethod
+    def pre_invocation_app_level(cls,
+                                 logger: Logger,
+                                 context: Context,
+                                 func_args: Dict[str, object] = {},
+                                 *args,
+                                 **kwargs) -> None:
+        """An implementation of pre invocation hooks on Function App's level.
+        The Python Worker Extension Interface is defined in
+        https://github.com/Azure/azure-functions-python-library/
+        blob/dev/azure/functions/extension/app_extension_base.py
+        """
+        if not cls._has_configure_called:
```

Comment:
What is the reason why `configure` must be called? Also for monitoring solutions, we don't want to crash the user's application on anything besides "on-startup", when the issue is a configuration issue (like missing ikey).

Reply:
Yeah, changed the code path to throw a warning instead of terminating the invocation.